### ISSUE TITLE
dxWDL ignores  default_runtime_attribtues.docker

### ISSUE THREAD

**Issue by jtratner** [dnanexus/dxWDL#313]
dxWDL seems to ignore `default_runtime_attributes.docker` in v1.31.  If you specify an extras, it will not run all commands within that image.  :(

To reproduce

```
docker run -e DX_SECURITY_CONTEXT=$(cat ~/.dnanexus_config/environment.json| jq -r .DX_SECURITY_CONTEXT) -v $PWD:/build -w /build dnanexus/dxwdl:v1.31 compile example.wdl -extras extras.json -project $PROJECT -overwrite
```

example.wdl:
```
version 1.0
task mytask {
    command <<<
    csvcut -h  #command only in image
    echo 5
    touch "out.txt"
    >>>
    output {
        File other = "out.txt"
    }
}
```

extras.json:
```
{
    "default_runtime_attributes" : {
      "docker" : "dx://<some image>"
    }
}

```

**Comment by orodeh** [comment]
Release v1.32 contains a fix for this issue. 

The problem is that in the 0.x series, we rewrote the WDL source code. This made it easy to  modify the runtime attributes, so we could slip in global runtime defaults if those were specified. 

In the 1.x series, we gave up on the idea of rewriting WDL, because it turned out to be too complex to get right. Corner cases kept creeping up, for example, what happens if you have quotes within quotes, or, special characters. During the move from 0.x to 1.x, I *think* the capability to use global defaults was lost. I guess nobody noticed until now. 

The more difficult part of the fix is to embed the global defaults into the applet `details` field. This allows jobs to consult the attributes at runtime. For example, if task `foo` doesn't have the docker property set in its WDL source code, the `TaskRunner` will also be able to check the global defaults at runtime.
